x-pack/plugins/enterprise_search/server/utils/create_ml_inference_pipeline.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IngestGetPipelineResponse, IngestPipeline } from '@elastic/elasticsearch/lib/api/types';
+import { ElasticsearchClient } from '@kbn/core/server';
+
+import { ErrorCode } from '../../common/types/error_codes';
+
+import { formatMlPipelineBody } from './create_pipeline_definitions';
+
+/**
+ * Details of a created or updated pipeline.
+ */
+export interface ModifiedPipeline {
```

Comment:
I'd recommend renaming this to `ModifiedPipelineResponse` to distinguish it as it's intention

Reply:
I'm not sure, that would create coupling between the application layer (we're returning the results of processing) and the API (we're returning an HTTP response).